✨ Feat: #41 RecordSaveSheetView 및 사진 선택 기능 구현

NOP-YA opened the pull request:
<!--
🙏 PR 제목 컨벤션 (Gitmoji + 타입 + 이슈 번호 + 작업 요약)
예시: ✨ Feat: #167 예약 취소 구현
※ PR 생성 시 Assignees 및 Labels 설정도 잊지 마세요!
-->

## ✨ What’s this PR?
### 📌 관련 이슈 (Related Issue)
<!-- 해당 PR이 어떤 이슈를 해결하는지 연결해주세요 -->
- Closes #41

---

### 🧶 주요 변경 내용 (Summary)
<!-- 이번 PR에서 작업한 핵심 변경 사항을 작성해주세요 -->

- RecordSaveSheetView 레이아웃 및 상태 구성
- 사용자 앨범의 최근 사진 최대 20장 미리보기 구현
- 미리보기 썸네일에서 최대 4장까지 사진 선택 가능
- 선택된 사진이 1장 이상일 때만 저장 버튼이 활성화됨
- 저장 버튼 탭 시 임시 저장 로직 실행
- 앨범 접근 버튼 추가 (※ 앨범 진입 후 기존 선택 반영 로직은 미구현)


---

### 📸 스크린샷 (Optional)
<!-- UI 작업의 경우, 구현한 화면을 첨부해주세요 -->
<!-- 이미지 크기 조절 예시: <img width="300" alt="설명" src="링크"> -->

<img width="300" alt="IMG_2609" src="https://github.com/user-attachments/assets/62636939-db7c-4791-93ea-6252adc66c05" />
<img width="300" alt="IMG_2610" src="https://github.com/user-attachments/assets/c96cf50f-3b94-472d-b68b-34997c68ee19" />


---

### 🧪 테스트 / 검증 내역
<!-- 동작 확인 여부나 시나리오 테스트 내용을 간단히 써주세요 -->

- [x] UI 정상 동작 확인
- [x] iPhone 15, iOS 17.4 환경에서 정상 동작
- [ ] 다크모드 테스트는 이후 이슈로 분리 예정 (#000)

---

### 💬 기타 공유 사항
<!-- 리뷰어가 참고하면 좋을 정보, 고민했던 지점 등을 적어주세요 -->

- 너무 졸린 상태로 하느라, 정리는 많이 못한거 같습니다
- 원래 향후, 데이터 연결할때를 고려해서 했어야 했는데 그부분 위주로 좀 봐주시면 좋을거 같습니다
- SwiftUIView는 테스트 용으로 넣어놓은 뷰인데, 뺴고 커밋했어야 했을까요?
---

### 🙇🏻‍♀️ 리뷰 가이드 (선택)
<!-- 리뷰어가 중점적으로 보면 좋을 포인트가 있다면 알려주세요 -->

- `XxxViewModel.swift` 파일의 로직 분리 구조
- `NetworkManager.swift`의 공통 로직 처리 방식

NOP-YA committed:
✨ Feat. RecordSaveSheetView 및 사진 선택 기능 구현

Why:
- 기록 저장 흐름을 구성하기 위한 초기 뷰와 사진 선택 UX가 필요함

How:
- RecordSaveSheetView 기본 레이아웃 및 상태 구조 설계
- 사용자 앨범의 최근 사진 최대 20장 미리보기 구현
- 미리보기에서 최대 4장까지 선택 가능하도록 제한 로직 적용
- 선택된 사진이 1장 이상일 때 저장 버튼 활성화
- 저장 버튼 UI 및 임시 저장 기능 구현
- 앨범 접근 버튼 추가

Tag: #feat #photos #record #sheet #swiftui

freeskyES commented on Packages/Features/Sources/Features/RecordSave/RecordSaveSheetView.swift:
```diff
@@ -0,0 +1,131 @@
+            SelectedFlowerCardView(
+                flowerName: "프리지아",
+                flowerMeaning: "영원한 사랑",
+                flowerImageName: "flower"
```
사용자가 어떤 꽃을 선택하느냐에 따라 달라지는 동적 데이터이기 때문에,
현재처럼 하드코딩하는 대신 뷰모델에서 전달받은 값으로 주입하는 방식이 더 적절할 것 같습니다.

예를 들어, 아래와 같은 구조체로 selectedFlower를 정의하고
```swift
struct SelectedFlower {
    let name: String
    let meaning: String
    let imageName: String
}
```
이렇게 바인딩하면 더 명확하고 재사용성도 좋아질 것 같아요
```swift
SelectedFlowerCardView(
    flowerName: viewModel.selectedFlower.name,
    flowerMeaning: viewModel.selectedFlower.meaning,
    flowerImageName: viewModel.selectedFlower.imageName
)
```

NOP-YA replied:
리뷰 감사합니다. 반영했습니다.

NOP-YA committed:
♻️ Refactor. RecordSaveSheetView 구조 및 비동기 흐름 개선 (#41)

Why:
- UI 및 ViewModel 간 역할 분리를 명확히 하고,
  최근 사진 비동기 로드 과정에서의 Race Condition 문제를 방지하기 위함

How:
- ViewModel에 주입되는 데이터 구조를 동적으로 개선하여 하드코딩 제거
- RecordSaveSheetView 내 UI 구성 개선 및 역할 정리
- fetchRecentPhotos 함수 내 이미지 배열 Race Condition 이슈를 해결하기 위해
  기존 DispatchGroup 기반에서 async/await 기반 구조로 전환

Tag: #refactor #async #ui #viewmodel #photos
